pkg/volume/plugins.go

```diff
@@ -119,6 +119,39 @@ type VolumePluginMgr struct {
 	plugins map[string]VolumePlugin
 }
 
+// InternalVolume is an internal representation of a volume.  All API volume types translate to InternalVolume.
+type InternalVolume struct {
```

Comment:
suggest `type Spec`

Any particular reason to break out VolumeSource?  I would just list them directly here.  It's simpler and it is self-contained.

```
type Spec struct {
    Name string
    // Exactly one of the following must be set.
    VolumeSource *api.VolumeSource
    PersistentVolumeSource *api.PersistentVolumeSource
}
```